File: VB/HowToShowWindowViaAction.Module/BusinessClasses.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace HowToShowWindowViaAction.Module
	<DefaultClassOptions> _
	Public Class BusinessClass1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private name_Renamed As String

		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
	End Class
	<DefaultClassOptions> _
	Public Class BusinessClass2
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private name_Renamed As String

		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/HowToShowWindowViaAction.Module/BusinessClasses.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace HowToShowWindowViaAction.Module {
	[DefaultClassOptions]
	public class BusinessClass1: BaseObject {
		public BusinessClass1(Session session)
			: base(session) {
		}
		private string name;

		public string Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
	}
	[DefaultClassOptions]
	public class BusinessClass2 : BaseObject {
		public BusinessClass2(Session session)
			: base(session) {
		}
		private string name;

		public string Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
	}
}
